src/Microsoft.Azure.WebJobs.Host/Executors/HostMessageExecutor.cs

```diff
@@ -124,7 +128,8 @@ private async Task ProcessCallAndOverrideMessage(CallAndOverrideMessage message,
 
             if (instance != null)
             {
-                await _innerExecutor.TryExecuteAsync(instance, cancellationToken);
+                Func<IFunctionInstance> instanceFactory = () => CreateFunctionInstance(message);
```

Comment:
Need to remove the CreateFunctionInstance call above. To do so, you'll hoist the _functionLookup.Lookup check out of that method, and change the if (instance != null) check to instead check that the function lookup didn't return null.